frontend/src/routes/+page.svelte

```diff
@@ -1,2 +1,194 @@
-<h1>This is where the main page is shown</h1>
-<h2>Hoverable tiles</h2>
+<script lang="ts">
+    import gsap from 'gsap';
+    import SplitType from 'split-type';
+    import { spring } from 'svelte/motion';
+
+    import { page } from '$app/stores';
+    import { goto } from '$app/navigation';
+
+    import Book from '$icons/Main/Book.svelte';
+    import LightBulb from '$icons/Main/LightBulb.svelte';
+    import Puzzle from '$icons/Main/Puzzle.svelte';
+    import Responsive from '$icons/Main/Responsive.svelte';
+    import Atom from '$icons/Main/Atom.svelte';
+    import MolecularStructure from '$icons/Main/MolecularStructure.svelte';
+    import Molecule from '$icons/Main/Molecule.svelte';
+    import RoundBeaker from '$icons/Main/RoundBeaker.svelte';
+    import ArrowDown from '$icons/Sidebar/ArrowDown.svelte';
+
+    // If the user is already logged in, redirect them to questions (for now)
+    $: {
+        if ($page.data.user) {
+            goto('/question/1');
+        }
+    }
+
+    /* ---- ANIMATION FOR TEXT ---- */
+
+    let animateTextWrapper: HTMLElement;
+    let animatePopIn: HTMLElement;
+
+    const animationDuration = 0.3;
+    let animationStarted = false;
+    let timeline: gsap.core.Timeline;
+
+    $: {
+        if (animateTextWrapper) {
+            Array.from(animateTextWrapper.children).forEach((elem) => {
+                timeline = gsap.timeline();
+                const splitText = new SplitType(elem as HTMLElement, {
+                    types: 'lines,words,chars'
+                });
+                timeline.from(splitText.words, {
+                    y: 10,
+                    opacity: 0,
+                    duration: animationDuration,
+                    stagger: 0.05
+                });
+                timeline.from(splitText.chars, {
+                    y: 10,
+                    opacity: 0,
+                    duration: animationDuration,
+                    stagger: 0.025
+                });
+            });
+        }
+        if (animatePopIn) {
+            timeline.fromTo(
+                animatePopIn,
+                {
+                    opacity: 0,
+                    scale: 0.1
+                },
+                {
+                    opacity: 1,
+                    scale: 1,
+                    duration: animationDuration * 0.5,
+                    stagger: 0.05
+                }
+            );
+        }
+    }
+
+    /* ---- ANIMATION FOR PARTICLES ---- */
+    const randomSpringCoords = () => {
+        const xMin = 10;
+        const xMax = 40;
+        const yMin = 0;
+        const yMax = 90;
+        const coords = spring({
+            x: Math.random() * (xMax - xMin) + xMin,
+            y: Math.random() * (yMax - yMin) + yMin
```

Comment:
Can you write a helper function for this? Line 87 and 88 uses the same logic.

Reply:
Yep will do